src/transformers/modeling_flax_utils.py

```diff
@@ -163,31 +180,43 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         # Instantiate model.
         with open(resolved_archive_file, "rb") as state_f:
             try:
-                from flax.serialization import from_bytes
-
-                state = from_bytes(cls.model_class, state_f)
-            except TypeError:
-                try:
+                if from_pt:
                     import torch
 
                     state = torch.load(state_f)
                     state = {k: v.numpy() for k, v in state.items()}
                     state = cls.convert_from_pytorch(state, config)
                     state = unflatten_dict({tuple(k.split(".")[1:]): v for k, v in state.items()})
-                except UnpicklingError:
-                    raise EnvironmentError(
-                        f"Unable to convert model {archive_file} to Flax deserializable object. "
-                        "Supported format are PyTorch archive or Flax msgpack"
-                    )
+                else:
+                    state = from_bytes(cls, state_f.read())
```

Comment:
think we need to add a `read()` statement here to make it work with `from_bytes`